_posts/2019-06-05-uncertainty-estimates-tfprobability/uncertainty_tfprobability.Rmd

```diff
@@ -0,0 +1,782 @@
+
+## What to expect from this post
+
+Starting from what _not_ to expect: There won't be a recipe that tells you how _exactly_ to set all parameters involved in order to report the "right" uncertainty measures. But then, what _are_ the "right" uncertainty measures? Unless you happen to work with a method that has no (hyper-)parameters to tweak, there will always be uncertainty about how to report uncertainty.
+
+What you _can_ expect though is an introduction to obtaining uncertainty estimates for Keras networks, as well as an empirical report of how tweaking (hyper-)parameters may affect the results. As in the aforementioned post, we perform our tests on a simulated as well as on a real dataset, the [Combined Cycle Power Plant Data Set](http://archive.ics.uci.edu/ml/datasets/Combined+Cycle+Power+Plant). At the end, in place of strict rules you should have acquired some intuition that will transfer to other real-world datasets.
+
+Did you notice our talking about _Keras networks_ above? Indeed this post has an additional goal: So far, we haven't really discussed yet how `tfprobability` goes together with `keras`. Now we finally do (in short: they work together seemlessly).
+
+Finally, the notions of _aleatoric_ and _epistemic_ uncertainty, which may have stayed a bit abstract in the prior post, should get much more concrete here.
+
+## Aleatoric vs. epistemic uncertainty
+
+Reminiscent somehow of the classic decomposition of generalization error into bias and variance, splitting uncertainty into its epistemic and aleatoric constituents separates an irreducible from a reducible part.
+
+The reducible part relates to imperfection in the model: In theory, if our model were perfect, epistemic uncertainty would vanish. Put differently, if the training data were unlimited -- or if they comprised the whole population -- we could just add capacity to the model until we've obtained a perfect fit.
+
+In contrast, normally there is variation in our measurements. There may be one true process that determines my resting heart rate; nonetheless, actual measurements will vary over time. There is nothing to be done about this: This is the aleatoric part that just remains, to be factored into our expectations.
+
+Now reading this, you might be thinking: "Wouldn't a model that actually _were_ perfect capture those pseudo-random fluctuations?". We'll leave that phisosophical question be;  instead, we'll try to illustrate the usefulness of this distinction by example, in a practical way. In a nutshell, viewing a model's _aleatoric_ uncertainty output should caution us to factor in appropriate deviations when making our predictions, while inspecting _epistemic_ uncertainty should help us re-think the appropriateness of the chosen model.
```

Comment:
```suggestion
Now reading this, you might be thinking: "Wouldn't a model that actually _was_ perfect capture those pseudo-random fluctuations?". We'll leave that philosophical question be;  instead, we'll try to illustrate the usefulness of this distinction by example, in a practical way. In a nutshell, viewing a model's _aleatoric_ uncertainty output should caution us to factor in appropriate deviations when making our predictions, while inspecting _epistemic_ uncertainty should help us rethink the appropriateness of the chosen model.
```

Reply:
^^ changes were to was (since model is singular), corrects spelling of philosophical and rethink.